### PR TITLE
feat: add asr endpoints, dockerfiles and helm chart

### PR DESCRIPTION
**Pull request by Phil91**
## Description

- add get endpoint for credentials
- add swagger
- add post endpoint to validate json schemas
- add dockerfiles, helm chart and environment configuration
- improve GH workflows
- improve docs

## Why

To enable implementation for SSI Authority & Schema Registry.

## Issue

#7  #11 #13 

Issue for Sonarcloud enabling: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4732

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/ssi-authority-schema-registry/blob/main/docs/developer/Technical-Documentation/Dev-Process/How-to-contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added copyright and license headers, footers (for .md files) or files (for images)

### CODE REVIEW

**Comment by evegufy** [comment]
IP Issues created (due to failing dependencies check):
A review is required for nuget/nuget/-/Testcontainers.PostgreSql/3.8.0.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15169
A review is required for nuget/nuget/-/FakeItEasy/8.2.0.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15170
A review is required for nuget/nuget/-/JsonSchema.Net/7.0.2.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15171
A review is required for nuget/nuget/-/JsonPointer.Net/5.0.0.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15172
A review is required for nuget/nuget/-/Json.More.Net/2.0.1.2.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15173
A review is required for nuget/nuget/-/xunit.core/2.8.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15174
A review is required for nuget/nuget/-/xunit.analyzers/1.14.0.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15175
A review is required for nuget/nuget/-/xunit.assert/2.8.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15176
A review is required for nuget/nuget/-/xunit.extensibility.core/2.8.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15177
A review is required for nuget/nuget/-/Testcontainers/3.8.0.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15178
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15179
A review is required for nuget/nuget/-/xunit.extensibility.execution/2.8.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15180
A review is required for nuget/nuget/-/xunit.runner.visualstudio/2.8.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15181

**Review by evegufy — CHANGES_REQUESTED** [review]
@Phil91 I'm done with all my changes, could you please update PR title and description?

File: src/database/SsiAuthoritySchemaRegistry.Migrations/Seeder/BatchInsertSeeder.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        logger.LogDebug("Start seeding {Filename}", fileName);
+        var additionalEnvironments = _settings.TestDataEnvironments ?? Enumerable.Empty<string>();
+        var data = await SeederHelper.GetSeedData<T>(logger, fileName, _settings.DataPaths, cancellationToken, additionalEnvironments.ToArray()).ConfigureAwait(false);
+        logger.LogDebug("Found {ElementCount} data", data.Count);
+        if (data.Any())
+        {
+            var typeName = typeof(T).Name;
+            logger.LogDebug("Started to Seed {TableName}", typeName);
+            data = data.GroupJoin(context.Set<T>(), keySelector, keySelector, (d, dbEntry) => new { d, dbEntry })
```
Comment (ntruchsess) [inline src/database/SsiAuthoritySchemaRegistry.Migrations/Seeder/BatchInsertSeeder.cs:65]: it's not a good style to override a reference if not required to do so. Creating new local references is cheap.

File: src/registry/SsiAuthoritySchemaRegistry.Service/BusinessLogic/SchemaBusinessLogic.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        var location = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        if (location == null)
+        {
+            throw new UnexpectedConditionException("Assembly location must be set");
+        }
+
+        var path = Path.Combine(location, "Schemas", $"{schemaType}.schema.json");
+        var schemaJson = await File.ReadAllTextAsync(path, cancellationToken).ConfigureAwait(false);
+
+        var schema = JsonSchema.FromText(schemaJson);
```
Comment (ntruchsess) [inline src/registry/SsiAuthoritySchemaRegistry.Service/BusinessLogic/SchemaBusinessLogic.cs:41]:
``` 
var schema = await JsonSchema.FromStream(File.openRead(path)).ConfigureAwait(ConfigureAwaitOptions.None);
```

File: src/registry/SsiAuthoritySchemaRegistry.Service/Controllers/RegistryController.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+using System.Diagnostics.CodeAnalysis;
+
+namespace Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.Service.Controllers;
+
+/// <summary>
+/// Creates a new instance of <see cref="RegistryController"/>
+/// </summary>
+public static class RegistryController
+{
+    [ExcludeFromCodeCoverage]
```
Comment (ntruchsess) [inline src/registry/SsiAuthoritySchemaRegistry.Service/Controllers/RegistryController.cs:34]: non-testing code should not be excluded from coverage

File: src/registry/SsiAuthoritySchemaRegistry.Service/Controllers/SchemaController.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+using Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.Service.Extensions;
+using Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.Service.Models;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+
+namespace Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.Service.Controllers;
+
+public static class SchemaController
+{
+    [ExcludeFromCodeCoverage]
```
Comment (ntruchsess) [inline src/registry/SsiAuthoritySchemaRegistry.Service/Controllers/SchemaController.cs:12]: non-testing code should not be excluded from coverage

File: src/registry/SsiAuthoritySchemaRegistry.Service/Extensions/RouteHandlerBuilderExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+using Org.Eclipse.TractusX.Portal.Backend.Framework.ErrorHandling.Web;
+using Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.Service.Models;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.Service.Extensions;
+
+[ExcludeFromCodeCoverage]
```
Comment (ntruchsess) [inline src/registry/SsiAuthoritySchemaRegistry.Service/Extensions/RouteHandlerBuilderExtensions.cs:26]: non-testing code should not be excluded from coverage

File: src/registry/SsiAuthoritySchemaRegistry.Service/Program.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+using Org.Eclipse.TractusX.Portal.Backend.Framework.Web;
+using Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.DbAccess.DependencyInjection;
+using Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.Service.Controllers;
+using System.Text.Json.Serialization;
+
+const string Version = "v1";
+
+WebApplicationBuildRunner
+    .BuildAndRunWebApplication<Program>(args, "registry", Version, ".Registry",
```
Comment (ntruchsess) [inline src/registry/SsiAuthoritySchemaRegistry.Service/Program.cs:28]: as await is supported in toplevel program.cs BuildAndRunWebApplication should be refactored as async and awaited here. (this saves a thread that otherwiese synchronously waits for program.cs to end).

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Comment by evegufy** [comment]
@Phil91 could you please also enable sonarcloud, see ssi issuer issue https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4403

**Review by evegufy — APPROVED** [review]
as discussed, please still update PR title and description

File: src/SsiAuthoritySchemaRegistry.sln.DotSettings.user [hunk 7]
```diff
@@ -0,0 +1,15 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/SsiAuthoritySchemaRegistry.sln.DotSettings.user]: removed

File: src/database/SsiAuthoritySchemaRegistry.DbAccess/DependencyInjection/RegistryRepositoriesServiceExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.Entities;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Org.Eclipse.TractusX.SsiAuthoritySchemaRegistry.DbAccess.DependencyInjection;
+
+public static class RegistryRepositoriesServiceExtensions
+{
+    [ExcludeFromCodeCoverage]
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/database/SsiAuthoritySchemaRegistry.DbAccess/DependencyInjection/RegistryRepositoriesServiceExtensions.cs]: removed

File: src/database/SsiAuthoritySchemaRegistry.Entities/RegistryContext.cs [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    {
+        optionsBuilder.UseSnakeCaseNamingConvention();
+    }
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.HasAnnotation("Relational:Collation", "en_US.utf8");
+        modelBuilder.HasDefaultSchema("ssi-authority-schema-registry");
+
+        modelBuilder.Entity<Authority>(e =>
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/database/SsiAuthoritySchemaRegistry.Entities/RegistryContext.cs]: adjusted

File: src/database/SsiAuthoritySchemaRegistry.Entities/RegistryContext.cs [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        {
+            e.HasKey(a => a.Bpn);
+        });
+
+        modelBuilder.Entity<Credential>(e =>
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/database/SsiAuthoritySchemaRegistry.Entities/RegistryContext.cs]: adjusted

File: src/database/SsiAuthoritySchemaRegistry.Migrations/Program.cs [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+}
+catch (Exception ex) when (!ex.GetType().Name.Equals("StopTheHostException", StringComparison.Ordinal))
+{
+    Log.Fatal("Unhandled exception {Exception}", ex);
+    throw;
+}
+finally
+{
+    Log.Information("Process Shutting down...");
+    Log.CloseAndFlush();
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/database/SsiAuthoritySchemaRegistry.Migrations/Program.cs]: adjusted

File: src/database/SsiAuthoritySchemaRegistry.Migrations/Seeder/BatchInsertSeeder.cs [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        await context.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
+    }
+
+    private async Task SeedTable<T>(string fileName, Func<T, object> keySelector, CancellationToken cancellationToken) where T : class
+    {
+        logger.LogDebug("Start seeding {Filename}", fileName);
+        var additionalEnvironments = _settings.TestDataEnvironments ?? Enumerable.Empty<string>();
+        var data = await SeederHelper.GetSeedData<T>(logger, fileName, _settings.DataPaths, cancellationToken, additionalEnvironments.ToArray()).ConfigureAwait(false);
+        logger.LogDebug("Found {ElementCount} data", data.Count);
+        if (data.Any())
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/database/SsiAuthoritySchemaRegistry.Migrations/Seeder/BatchInsertSeeder.cs]: adjusted.

File: src/database/SsiAuthoritySchemaRegistry.Migrations/Seeder/BatchInsertSeeder.cs (hunk 1 above)
Comment (Phil91) [reply src/database/SsiAuthoritySchemaRegistry.Migrations/Seeder/BatchInsertSeeder.cs]: adjusted

File: src/registry/SsiAuthoritySchemaRegistry.Service/BusinessLogic/SchemaBusinessLogic.cs (hunk 2 above)
Comment (Phil91) [reply src/registry/SsiAuthoritySchemaRegistry.Service/BusinessLogic/SchemaBusinessLogic.cs]: adjusted

File: src/registry/SsiAuthoritySchemaRegistry.Service/Controllers/RegistryController.cs (hunk 3 above)
Comment (Phil91) [reply src/registry/SsiAuthoritySchemaRegistry.Service/Controllers/RegistryController.cs]: removed

File: src/registry/SsiAuthoritySchemaRegistry.Service/Controllers/SchemaController.cs (hunk 4 above)
Comment (Phil91) [reply src/registry/SsiAuthoritySchemaRegistry.Service/Controllers/SchemaController.cs]: removed

File: src/registry/SsiAuthoritySchemaRegistry.Service/Extensions/RouteHandlerBuilderExtensions.cs (hunk 5 above)
Comment (Phil91) [reply src/registry/SsiAuthoritySchemaRegistry.Service/Extensions/RouteHandlerBuilderExtensions.cs]: removed

File: src/registry/SsiAuthoritySchemaRegistry.Service/Program.cs (hunk 6 above)
Comment (Phil91) [reply src/registry/SsiAuthoritySchemaRegistry.Service/Program.cs:28]: will adjust it as soon as https://github.com/eclipse-tractusx/portal-backend/pull/786 is merged and the new packages are available.

Comment (Phil91) [reply src/registry/SsiAuthoritySchemaRegistry.Service/Program.cs]: adjusted

**Review by evegufy — APPROVED** [review]

File: src/registry/SsiAuthoritySchemaRegistry.Service/appsettings.json [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+      "Application": "SsiAutoritySchemaRegistry"
+    }
+  },
+  "SwaggerEnabled": "",
+  "HealthChecks": [],
+  "Cors": {
+    "AllowedOrigins": []
+  },
+  "ConnectionStrings": {
+    "RegistryDb": "Server=placeholder;Database=placeholder;Port=5432;User Id=placeholder;Password=;Ssl Mode=Disable;"
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/registry/SsiAuthoritySchemaRegistry.Service/appsettings.json:30]: there is no password configured?

**Review by evegufy — APPROVED** [review]

**Pull request merged**